src/vcpkg/commands.z-changelog.cpp

```diff
@@ -0,0 +1,114 @@
+
+#include <vcpkg/commands.portsdiff.h>
+#include <vcpkg/commands.z-changelog.h>
+#include <vcpkg/vcpkgcmdarguments.h>
+#include <vcpkg/vcpkgpaths.h>
+
+namespace vcpkg
+{
+    constexpr CommandMetadata CommandZChangelogMetadata{
+        "z-changelog",
+        "Generate github.com/microsoft/vcpkg changelog",
+        {},
+        Undocumented,
+        AutocompletePriority::Never,
+        1,
+        1,
+        {},
+        nullptr,
+    };
+
+    void command_z_changelog_and_exit(const VcpkgCmdArguments& args, const VcpkgPaths& paths)
+    {
+        const auto parsed = args.parse_arguments(CommandZChangelogMetadata);
+        const StringView git_commit_id_for_previous_snapshot = parsed.command_arguments[0];
+        const auto portsdiff = find_portsdiff(paths, git_commit_id_for_previous_snapshot, "HEAD");
+
+        std::string result;
+        result.append("vcpkg (FROM - TO)\n");
+        result.append("---\n");
+        auto total_port_count = paths.get_filesystem()
+                                    .get_directories_non_recursive(paths.builtin_ports_directory(), VCPKG_LINE_INFO)
+                                    .size();
+        result.append(fmt::format("#### Total port count: {}\n", total_port_count));
+        result.append("#### Total port count per triplet (tested) (tentative): LINK\n");
+        result.append("|triplet|ports available|\n");
+        result.append("|---|---|\n");
+        result.append("|x86-windows|NUMBER|\n");
+        result.append("|**x64-windows**|NUMBER|\n");
+        result.append("|x64-windows-static|NUMBER|\n");
+        result.append("|x64-windows-static-md|NUMBER|\n");
+        result.append("|x64-uwp|NUMBER|\n");
+        result.append("|arm64-windows|NUMBER|\n");
+        result.append("|arm64-uwp|NUMBER|\n");
+        result.append("|**x64-osx**|NUMBER|\n");
+        result.append("|**x64-linux**|NUMBER|\n");
+        result.append("|arm-neon-android|NUMBER|\n");
+        result.append("|x64-android|NUMBER|\n");
+        result.append("|arm64-android|NUMBER|\n");
+        result.append("\n");
+
+        result.append("The following vcpkg-tool releases have occurred since the last registry release:\n");
+        result.append("* \n");
+        result.append("\n");
+
+        result.append("In those tool releases, the following changes are particularly meaningful:\n");
+        result.append("* \n");
+        result.append("\n");
+
+        if (!portsdiff.added_ports.empty())
+        {
+            result.append("<details>\n");
+            result.append(fmt::format("<summary><b>The following {} ports have been added:</b></summary>\n\n",
+                                      portsdiff.added_ports.size()));
+            result.append("|port|version|\n");
+            result.append("|---|---|\n");
+            for (auto&& added_port : portsdiff.added_ports)
+            {
+                result.append(fmt::format("|{}|{}|\n", added_port.port_name, added_port.version));
```

Comment:
Is there a reasonable way to use `fmt::format_to()` and a back_inserter here? Something like
```suggestion
                // out_it defined as `auto out_it = std::back_inserter(result);`
                fmt::format_to(out_it, "|{}|{}|\n", added_port.port_name, added_port.version);
```

Reply:
Yes, but perf super doesn't matter and I was lazy. Stand by...

Reply:
Done!